### PR TITLE
AKU-693: Ensure options provided by ServiceStore handle correct scoping

### PR DESCRIPTION
**Pull request by draperd**
This PR addresses https://issues.alfresco.com/jira/browse/AKU-693. It resolves the issue of correctly scoping option request handling.

### CODE REVIEW

File: aikau/src/test/resources/testApp/js/aikau/testing/mockservices/SearchMockXhr.js [hunk 1]
```diff
@@ -34,21 +40,66 @@ define(["dojo/_base/declare",
...
+         var url = request.url;
+         var query = url.substring(url.indexOf("?") + 1);
+         var queryObj = ioQuery.queryToObject(query);
+         if (queryObj.term === "hame" && queryObj.spellcheck === "false")
+         {
+            request.respond(200, {
+               "Content-Type": "application/json;charset=UTF-8"
+            }, original);
+         }
+         else if (queryObj.term === "hame" && queryObj.spellcheck === "true")
```
Comment (MartinDoyleUK) [inline aikau/src/test/resources/testApp/js/aikau/testing/mockservices/SearchMockXhr.js]: should this be `"home"`?

File: aikau/src/test/resources/testApp/js/aikau/testing/mockservices/SearchMockXhr.js [hunk 2]
```diff
@@ -34,21 +40,66 @@ define(["dojo/_base/declare",
...
+
+      /**
+       * Simulates getting the tags based on a supplied filter
+       *
+       * @instance
+       */
+      onAutoSuggest: function alfresco_testing_SearchMockXhr__onAutoSuggest(request) {
+         request.respond(200, {
+            "Content-Type": "application/json;charset=UTF-8"
+         }, suggestions);
```
Comment (MartinDoyleUK) [inline aikau/src/test/resources/testApp/js/aikau/testing/mockservices/SearchMockXhr.js]: `suggestions` will be `undefined`?

File: aikau/src/test/resources/testApp/WEB-INF/classes/alfresco/site-webscripts/alfresco/search/NoHashSearching.get.js [hunk 3]
```diff
@@ -9,7 +9,8 @@ model.jsonModel = {
             }
          }
       },
-      "aikauTesting/mockservices/SearchMockService",
+      // "aikauTesting/mockservices/SearchMockService",
```
Comment (MartinDoyleUK) [inline aikau/src/test/resources/testApp/WEB-INF/classes/alfresco/site-webscripts/alfresco/search/NoHashSearching.get.js]: Delete commented-out code?

File: aikau/src/main/resources/alfresco/forms/controls/utilities/ServiceStore.js [hunk 4]
```diff
@@ -300,8 +300,8 @@ define(["dojo/_base/declare",
          payload.query = query[this.queryAttribute || "name"];
 
          var optionsHandle = [];
-         optionsHandle.push(this.alfSubscribe(responseTopic + "_SUCCESS", lang.hitch(this, "onQueryOptions", response, query, resultsProperty, optionsHandle)));
-         optionsHandle.push(this.alfSubscribe(responseTopic, lang.hitch(this, "onQueryOptions", response, query, resultsProperty, optionsHandle)));
+         optionsHandle.push(this.alfSubscribe(responseTopic + "_SUCCESS", lang.hitch(this, "onQueryOptions", response, query, resultsProperty, optionsHandle), this.publishGlobal));
+         optionsHandle.push(this.alfSubscribe(responseTopic, lang.hitch(this, "onQueryOptions", response, query, resultsProperty, optionsHandle), this.publishGlobal));
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/forms/controls/utilities/ServiceStore.js]: `this.onQueryOptions` (and line above)?

File: aikau/src/test/resources/testApp/js/aikau/testing/mockservices/SearchMockXhr.js (hunk 1 above)
Comment (draperd) [inline aikau/src/test/resources/testApp/js/aikau/testing/mockservices/SearchMockXhr.js]: Lol... no, that's the point of the test - it's testing that alternative search suggestions are provided when you type something incorrectly, e.g. you enter "hame" but it automatically searches for "home" and tells you - but you can tell it to search for "hame" if you really meant it!

File: aikau/src/test/resources/testApp/js/aikau/testing/mockservices/SearchMockXhr.js (hunk 2 above)
Comment (draperd) [inline aikau/src/test/resources/testApp/js/aikau/testing/mockservices/SearchMockXhr.js]: Weird - that's not what it looks like locally, how odd.

Comment (draperd) [inline aikau/src/test/resources/testApp/js/aikau/testing/mockservices/SearchMockXhr.js]: Ah, I'd failed to stage the new file

File: aikau/src/test/resources/testApp/js/aikau/testing/mockservices/SearchMockXhr.js (hunk 1 above)
Comment (MartinDoyleUK) [inline aikau/src/test/resources/testApp/js/aikau/testing/mockservices/SearchMockXhr.js]: Worth checking :)

**Pull request closed without merging**